plugins/modules/ndo_route_map_policy_multicast.py

```diff
@@ -188,64 +189,47 @@ def main():
     mso_template.validate_template("tenantPolicy")
 
     path = "/tenantPolicyTemplate/template/mcastRouteMapPolicies"
-    existing_route_map_policies = mso_template.template.get("tenantPolicyTemplate", {}).get("template", {}).get("mcastRouteMapPolicies", [])
-    if route_map_policy:
-        object_description = "Multicast Route Map Policy"
-        if route_map_policy_uuid:
-            match = mso_template.get_object_by_uuid(object_description, existing_route_map_policies, route_map_policy_uuid)
-        else:
-            kv_list = [KVPair("name", route_map_policy)]
-            match = mso_template.get_object_by_key_value_pairs(object_description, existing_route_map_policies, kv_list)
-        if match:
-            match.details["entries"] = match.details.pop("mcastRtMapEntryList")
-            mso.existing = mso.previous = copy.deepcopy(match.details)
-    else:
-        mso.existing = mso.previous = existing_route_map_policies
+    match = get_multicast_route_map_policy(mso_template, route_map_policy_uuid, route_map_policy)
 
-    if state == "present":
-
-        if match:
+    if route_map_policy_uuid or route_map_policy:
+        if match:  # Query a specific object
+            mso.existing = copy.deepcopy(match.details)
+            mso.previous = copy.deepcopy(match.details)
```

Comment:
Simplify it ike this:

```suggestion
            mso.existing = mso.previous = copy.deepcopy(match.details)

```